build_runner/lib/src/generate/build_impl.dart

```diff
@@ -469,20 +469,23 @@ class _SingleBuild {
 
     // We only check the first output, because all outputs share the same inputs
     // and invalidation state.
-    if (!node.needsUpdate) return false;
+    if (node.state == GeneratedNodeState.upToDate) return false;
+    // Early bail out condition, this is a forced update.
+    if (node.state == GeneratedNodeState.definitelyNeedsUpdate) return true;
     // TODO: Don't assume the worst for globs
     // https://github.com/dart-lang/build/issues/624
-    if (node.previousInputsDigest == null || node.globs.isNotEmpty) {
+    if (node.previousInputsDigest == null) {
```

Comment:
ah, I guess we could have also considered clearing out the previous digest to indicate a glob change...

I think I like the enum better.

Reply:
ack